documentation/general/trust-code.md

```diff
@@ -0,0 +1,19 @@
+# Overview
+The .NET SDK is intended to be a collection of tools for manipulating, building, and executing code. As a result, there is no way to ensure that it is 100% secure regardless of how it is used. This document attempts to make clear the line between what security threats are up to the user and what threats the SDK should protect against. It defines the trust boundary where the user can assume that as long as they are secure in the ways outlined in this document, they will not be compromised by an outside threat actor.
+
+# Types of Threats
+
+## Untrusted Source Code
+You should never build or run code from an untrusted source without employing mitigating factors like those described [here](https://github.com/dotnet/sdk/blob/main/documentation/general/ExecutingCustomerCode.md). Additionally, some SDK commands will automatically find code or code-adjacent files in the current directory, project directory, executable directory, or any parent directory of any of those up to the file system root. Files that may affect a build or other command if found include but are not limited to a Directory.Build.props, global.json, NuGet.config, or dotnet-tools.json. You should also check your home folder as well as environment variables that may point to other locations the SDK will search to find code or configuration files.
```

Comment:
nit: Change the link to `[here](general/ExecutingCustomerCode.md)`. This doc itself lives in the same repo to the doc you're linking to, so always use a relative link. It makes a difference as then this doc always links to its sibling doc with the same branch. Plus, `main` isn't always guaranteed to exist, as that similar situation happened during the switch from `master` to `main`. StackOverflow on linking [here](https://stackoverflow.com/a/7658676/294804).